Library/Homebrew/cmd/fetch.rb

```diff
@@ -32,7 +32,7 @@ def fetch
   def fetch_bottle? f
     return true if ARGV.force_bottle? && f.bottle
     return false unless f.bottle && f.pour_bottle?
-    if f.file_modified?
+    if f.file_modified? && !ARGV.build_from_source?
```

Comment:
Will check this out later today. It does look like the logical option though, without having to chuck more code on the pile.